lib/core/base/audit.js

```diff
@@ -651,19 +675,16 @@ Audit.prototype.normalizeOptions = function(options) {
 		if (['rule', 'rules'].includes(only.type)) {
 			only.type = 'rule';
 			only.values.forEach(function(ruleId) {
-				if (!audit.getRule(ruleId)) {
+				if (!ruleIds.includes(ruleId)) {
 					throw new Error('unknown rule `' + ruleId + '` in options.runOnly');
 				}
 			});
 
 			// Validate 'tags' (e.g. anything not 'rule')
 		} else if (['tag', 'tags', undefined].includes(only.type)) {
 			only.type = 'tag';
-			const unmatchedTags = audit.rules.reduce((unmatchedTags, rule) => {
-				return unmatchedTags.length
-					? unmatchedTags.filter(tag => !rule.tags.includes(tag))
-					: unmatchedTags;
-			}, only.values);
+			only.type = 'tag';
```

Comment:
Why do this twice?

Reply:
Good question. Must have been a bad copy paste.